### ISSUE TITLE
Current nightly is broken. Use 2017-03-02

### ISSUE THREAD

**Issue by Doaxan** [Drakulix/fireplace#37]
Cannot build fireplace on fedora 25.
nightly-x86_64-unknown-linux-gnu (default)
rustc 1.17.0-nightly (b1e31766d 2017-03-03)

```
       Fresh wlc-sys v0.0.8
   Compiling wlc v2.0.1
     Running `rustc --crate-name wlc /home/doaxan/.cargo/registry/src/github.com-1ecc6299db9ec823/wlc-2.0.1/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 --cfg 'feature="serde_derive"' --cfg 'feature="static"' --cfg 'feature="render"' --cfg 'feature="serde"' --cfg 'feature="default"' --cfg 'feature="wlc-sys"' --cfg 'feature="serialization"' -C metadata=793a3ba1ed16e543 -C extra-filename=-793a3ba1ed16e543 --out-dir /home/doaxan/fireplace/target/release/deps -L dependency=/home/doaxan/fireplace/target/release/deps --extern log=/home/doaxan/fireplace/target/release/deps/liblog-1de695589fd5a905.rlib --extern num_traits=/home/doaxan/fireplace/target/release/deps/libnum_traits-f44a656d09f2ee1a.rlib --extern serde_derive=/home/doaxan/fireplace/target/release/deps/libserde_derive-e7c3727d54fc8c73.so --extern xkbcommon=/home/doaxan/fireplace/target/release/deps/libxkbcommon-c6d3b0dc4afe8d2d.rlib --extern lazy_static=/home/doaxan/fireplace/target/release/deps/liblazy_static-2c238253f1db6b80.rlib --extern libc=/home/doaxan/fireplace/target/release/deps/liblibc-56def1a5449a1812.rlib --extern serde=/home/doaxan/fireplace/target/release/deps/libserde-908251938cd969ac.rlib --extern bitflags=/home/doaxan/fireplace/target/release/deps/libbitflags-5601568f0bbd601b.rlib --extern wlc_sys=/home/doaxan/fireplace/target/release/deps/libwlc_sys-67d72ed26884df02.rlib --extern uinput_sys=/home/doaxan/fireplace/target/release/deps/libuinput_sys-feae97c3ba859673.rlib --cap-lints allow -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/lib -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/lib64 -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/build/protos -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/build/lib/chck/lib`

thread 'rustc' has overflowed its stack
fatal runtime error: stack overflow
error: Could not compile `wlc`.

Caused by:
  process didn't exit successfully: `rustc --crate-name wlc /home/doaxan/.cargo/registry/src/github.com-1ecc6299db9ec823/wlc-2.0.1/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 --cfg feature="serde_derive" --cfg feature="static" --cfg feature="render" --cfg feature="serde" --cfg feature="default" --cfg feature="wlc-sys" --cfg feature="serialization" -C metadata=793a3ba1ed16e543 -C extra-filename=-793a3ba1ed16e543 --out-dir /home/doaxan/fireplace/target/release/deps -L dependency=/home/doaxan/fireplace/target/release/deps --extern log=/home/doaxan/fireplace/target/release/deps/liblog-1de695589fd5a905.rlib --extern num_traits=/home/doaxan/fireplace/target/release/deps/libnum_traits-f44a656d09f2ee1a.rlib --extern serde_derive=/home/doaxan/fireplace/target/release/deps/libserde_derive-e7c3727d54fc8c73.so --extern xkbcommon=/home/doaxan/fireplace/target/release/deps/libxkbcommon-c6d3b0dc4afe8d2d.rlib --extern lazy_static=/home/doaxan/fireplace/target/release/deps/liblazy_static-2c238253f1db6b80.rlib --extern libc=/home/doaxan/fireplace/target/release/deps/liblibc-56def1a5449a1812.rlib --extern serde=/home/doaxan/fireplace/target/release/deps/libserde-908251938cd969ac.rlib --extern bitflags=/home/doaxan/fireplace/target/release/deps/libbitflags-5601568f0bbd601b.rlib --extern wlc_sys=/home/doaxan/fireplace/target/release/deps/libwlc_sys-67d72ed26884df02.rlib --extern uinput_sys=/home/doaxan/fireplace/target/release/deps/libuinput_sys-feae97c3ba859673.rlib --cap-lints allow -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/lib -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/lib64 -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/build/protos -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/build/lib/chck/lib` (exit code: 1)
```

**Comment by Drakulix** [comment]
I know, current nightly is broken. I am waiting for a new one, because new newest is from `03-03-2017` and we did not get a new one now since 6 days, so they probably know it's broken. If the next nightly does not fix this, I will open a bug at rust's repository.

In the meantime use this in fireplace build directory:
`rustup override set nightly-2017-03-02`

**Comment by Drakulix** [comment]
Or use the pre-build binary: https://github.com/Drakulix/fireplace/releases

**Comment by Doaxan** [comment]
I updated rust, but still get this error:
```active toolchain
----------------

nightly-x86_64-unknown-linux-gnu (default)
rustc 1.17.0-nightly (824c9ebbd 2017-03-12)```

```
       Fresh wlc-sys v0.0.8
   Compiling wlc v2.0.1
     Running `rustc --crate-name wlc /home/doaxan/.cargo/registry/src/github.com-1ecc6299db9ec823/wlc-2.0.1/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 --cfg 'feature="serde_derive"' --cfg 'feature="default"' --cfg 'feature="wlc-sys"' --cfg 'feature="serialization"' --cfg 'feature="static"' --cfg 'feature="serde"' --cfg 'feature="render"' -C metadata=793a3ba1ed16e543 -C extra-filename=-793a3ba1ed16e543 --out-dir /home/doaxan/fireplace/target/release/deps -L dependency=/home/doaxan/fireplace/target/release/deps --extern serde=/home/doaxan/fireplace/target/release/deps/libserde-908251938cd969ac.rlib --extern xkbcommon=/home/doaxan/fireplace/target/release/deps/libxkbcommon-c6d3b0dc4afe8d2d.rlib --extern serde_derive=/home/doaxan/fireplace/target/release/deps/libserde_derive-e7c3727d54fc8c73.so --extern bitflags=/home/doaxan/fireplace/target/release/deps/libbitflags-5601568f0bbd601b.rlib --extern log=/home/doaxan/fireplace/target/release/deps/liblog-1de695589fd5a905.rlib --extern wlc_sys=/home/doaxan/fireplace/target/release/deps/libwlc_sys-67d72ed26884df02.rlib --extern lazy_static=/home/doaxan/fireplace/target/release/deps/liblazy_static-2c238253f1db6b80.rlib --extern num_traits=/home/doaxan/fireplace/target/release/deps/libnum_traits-f44a656d09f2ee1a.rlib --extern libc=/home/doaxan/fireplace/target/release/deps/liblibc-56def1a5449a1812.rlib --extern uinput_sys=/home/doaxan/fireplace/target/release/deps/libuinput_sys-feae97c3ba859673.rlib --cap-lints allow -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/lib -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/lib64 -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/build/protos -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/build/lib/chck/lib`
error: Could not compile `wlc`.

Caused by:
  process didn't exit successfully: `rustc --crate-name wlc /home/doaxan/.cargo/registry/src/github.com-1ecc6299db9ec823/wlc-2.0.1/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 --cfg feature="serde_derive" --cfg feature="default" --cfg feature="wlc-sys" --cfg feature="serialization" --cfg feature="static" --cfg feature="serde" --cfg feature="render" -C metadata=793a3ba1ed16e543 -C extra-filename=-793a3ba1ed16e543 --out-dir /home/doaxan/fireplace/target/release/deps -L dependency=/home/doaxan/fireplace/target/release/deps --extern serde=/home/doaxan/fireplace/target/release/deps/libserde-908251938cd969ac.rlib --extern xkbcommon=/home/doaxan/fireplace/target/release/deps/libxkbcommon-c6d3b0dc4afe8d2d.rlib --extern serde_derive=/home/doaxan/fireplace/target/release/deps/libserde_derive-e7c3727d54fc8c73.so --extern bitflags=/home/doaxan/fireplace/target/release/deps/libbitflags-5601568f0bbd601b.rlib --extern log=/home/doaxan/fireplace/target/release/deps/liblog-1de695589fd5a905.rlib --extern wlc_sys=/home/doaxan/fireplace/target/release/deps/libwlc_sys-67d72ed26884df02.rlib --extern lazy_static=/home/doaxan/fireplace/target/release/deps/liblazy_static-2c238253f1db6b80.rlib --extern num_traits=/home/doaxan/fireplace/target/release/deps/libnum_traits-f44a656d09f2ee1a.rlib --extern libc=/home/doaxan/fireplace/target/release/deps/liblibc-56def1a5449a1812.rlib --extern uinput_sys=/home/doaxan/fireplace/target/release/deps/libuinput_sys-feae97c3ba859673.rlib --cap-lints allow -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/lib -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/lib64 -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/build/protos -L native=/home/doaxan/fireplace/target/release/build/wlc-sys-ecb5004244790a3f/out/build/lib/chck/lib` (exit code: 1)
```

**Comment by Drakulix** [comment]
I will report this upstream, thanks for the heads up.

**Comment by Drakulix** [comment]
Done: https://github.com/rust-lang/rust/issues/40493

Lets see what happens.

**Comment by Drakulix** [comment]
The issue to track seems to be https://github.com/rust-lang/rust/issues/40573. In the meantime you can fix your build by using `RUST_MIN_STACK=32000000` as an environment variable. I have also applied this workaround to the vagga and travis configuration files. So build with vagga should work out-of-the-box and we can again test and continue development.

I am leaving this open until the error is fixed upstream.
